src/ApplicationInsights.ServiceFabric.Native/Net45/IWrappingClient.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Microsoft.ApplicationInsights.ServiceFabric.Remoting.Activities
+{
+    using Microsoft.ServiceFabric.Services.Remoting.Client;
+
+    /// <summary>
+    /// Interface implemented by remoting clients to provide access to the inner client.
+    /// </summary>
+    internal interface IWrappingClient
```

Comment:
Only our CorrelatingServiceRemotingClient objects has the concept of an InnerClient and implements this interface. Our factory uses this to access the inner client. An alternative is for the factory to try downcasting the client to CorrelatingServiceRemotingClient, then access the InnerClient. Depending on your choice of design.